### ISSUE TITLE
Allow CompositeController to use owner references instead of a selector

### ISSUE THREAD

**Issue by grantr** [GoogleCloudPlatform/metacontroller#104]
I have a use case for controlling a resource whose definition isn't owned by me and has no `spec.selector` field. There will only ever be one or two child resources, so a selector isn't really necessary. DecoratorController won't work because it doesn't allow me to update `status`. CompositeController won't work because it requires a `spec.selector` field.

What I want is a hybrid of the two strategies: The capabilities of CompositeController with the owner referencing strategy of DecoratorController.

Would it be possible to extend CompositeController with an option to allow using owner references for tracking child objects instead of a selector?

**Comment by enisoc** [comment]
If you enable [selector generation](https://metacontroller.app/api/compositecontroller/#generate-selector) in CompositeController, it should be effectively the same as not using selectors at all. Metacontroller will not look for (nor set) a `spec.selector` in each parent object, and even if one is there, it will be ignored.

To be clear, CompositeController does already use owner references in addition to selectors, so if you "disable" selectors like above, you will essentially be using only owner references.

Does that work for your use case?

**Comment by enisoc** [comment]
Metacontroller "imagines" a selector on the parent, without trying to materialize it into `spec` (which by convention is only for inputs to a controller, so a controller should not modify its own spec as an output).

In the doc section I linked above, "the controller" refers to the control loop within Metacontroller that's doing your bidding, based on what you say you want via hook responses.

In that same section, "this behavior" refers to the following set of commands that you're effectively giving to Metacontroller by enabling `generateSelector`:

* When creating children for me, add a label that points to the parent object's unique ID.
* Don't expect a `spec.selector` in each parent object, and ignore it if it's there.
* Manage children as if each parent object had an imaginary label selector that selects on the unique ID label that was added to its children.

Thanks for the feedback on the docs. I'll try to incorporate some of the info above to make that section clearer.
